corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -101,13 +108,27 @@ def get_docs(self, from_db, startkey, last_doc_pk=None, limit=500):
         """
         raise NotImplementedError
 
+    def get_doc_count(self, from_db):
+        """Get the doc count from the given DB
+        :param from_db: The DB alias to query
+        """
+        sql_query = "SELECT reltuples FROM pg_class WHERE oid = '{}'::regclass"
```

Comment:
This relies on the autovacuum daemon which updates the stats regularly: https://wiki.postgresql.org/wiki/Count_estimate

This will return an estimate of the row count.